### PR TITLE
Move pytest and other package installation to make

### PR DESCRIPTION
**Pull request by annuay-google**
### Issue
https://b.corp.google.com/issues/356788168

On develop branch, the `pytest-check` pre-commit hook fails with the following message for new workstations:
`Executable pytest not found`

7DXJcGGbpkBzyd7

### Approach

1. Install pytest and other packages required for running tests in `install-dev-deps` make recipe
2. Run the pytest module as an executable directly using `python -m`, instead of relying on pytest executable being in the shell path. This addresses the problem of missing executable

### Testing

Provisioned a new workstation, ran `make install-dev-deps` followed by `pre-commit run --all-files`. Verified all hooks passed.

Screenshot 2024-08-08 at 9 05 26 PM

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,6 +69,9 @@ install-dev-deps: warn-terraform-version warn-packer-version check-pre-commit ch
 	go install mvdan.cc/sh/v3/cmd/shfmt@latest
 	go install golang.org/x/tools/cmd/goimports@latest
 	go install honnef.co/go/tools/cmd/staticcheck@latest
+	pip install -r community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/requirements.txt
+	pip install -r community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/tests/requirements.txt
+	pip install pytest
```
Comment (mr0re1) [inline Makefile:74]: `pytest` is already a part of `scripts/tests/requirements.txt`, why did you add it individually?

**Comment by mr0re1** [comment]
LGTM, please address the comment.

**Review by mr0re1 — APPROVED** [review]

**Comment by annuay-google** [comment]
> LGTM, please address the comment.

Fixed by removing `pip install pytest` in makefile. Thanks for pointing out, I hadn't noticed it in requirements.txt

File: Makefile (hunk 1 above)
Comment (annuay-google) [reply Makefile]: Removed. I hadn't noticed `pytest` being already present in the file
